File: app/lib/models/purchase_request.dart
class PurchaseRequest {
  final String fromUserId;
  final String toUserId;
  final String type;
  final String itemDescription;
  final String status;
  final String paymentId;
  final DateTime createdAt;

  PurchaseRequest({
    required this.fromUserId,
    required this.toUserId,
    required this.type,
    required this.itemDescription,
    required this.status,
    required this.paymentId,
    required this.createdAt,
  });

  factory PurchaseRequest.fromJson(Map<String, dynamic> json) =>
      PurchaseRequest(
        fromUserId: json['from_user_id'],
        toUserId: json['to_user_id'],
        type: json['type'],
        itemDescription: json['item_description'],
        status: json['status'],
        paymentId: json['payment_id'],
        createdAt: DateTime.parse(json['created_at']),
      );

  Map<String, dynamic> toJson() => {
        'from_user_id': fromUserId,
        'to_user_id': toUserId,
        'type': type,
        'item_description': itemDescription,
        'status': status,
        'payment_id': paymentId,
        'created_at': createdAt.toIso8601String(),
      };
}

File: functions/entity/purchase_request.js
// purchase_requests 컬렉션 (구매/예약 대행 요청)
// 문서 ID: 자동 생성
// {
//   from_user_id: string,
//   to_user_id: string,
//   type: string, // ticket_purchase, restaurant_reservation 등
//   item_description: string,
//   status: 'pending' | 'accepted' | 'rejected' | 'completed',
//   payment_id: string,
//   created_at: timestamp
// }

class PurchaseRequest {
  constructor(data) {
    this.from_user_id = data.from_user_id;
    this.to_user_id = data.to_user_id;
    this.type = data.type;
    this.item_description = data.item_description;
    this.status = data.status; // 'pending' | 'accepted' | 'rejected' | 'completed'
    this.payment_id = data.payment_id;
    this.created_at = data.created_at;
  }
}

module.exports = PurchaseRequest;
